internal/docstore/driver/driver.go

```diff
@@ -71,3 +75,28 @@ type Mod struct {
 	FieldPath []string
 	Value     interface{}
 }
+
+// A Query defines a query operation to find documents within a collection based
+// on a set of requirements.
+type Query struct {
+	FieldPaths []string         // the selected field paths
+	Filters    []Filter         // a list of filter for the query
+	Limit      int              // the number of result in one query request
+	StartAt    interface{}      // define the starting point of the query
+	Iter       DocumentIterator // the iterator for Get action
+}
+
+// A Filter defines a filter expression used to filter the query result.
+type Filter struct {
+	Field string      // the field path to filter
+	Op    string      // the operation, supports =, >, >=, <, <=
+	Value interface{} // the value to filter
+}
+
+// A DocumentIterator is an iterator that can iterator through the results (for Get
+// action), and returns the next page that can be passed into a Query as a
+// starting point.
+type DocumentIterator interface {
```

Comment:
Why have an iterator for the driver? For blob, we exposed an iterator on the portable type, but the driver method is "get next page".

Reply:
In Firestore, querying is a streaming RPC.

The Go mongoDB driver exposes an iterator.

Reply:
But, it is easy to implement returning pages on top of an iterator. It is hard to implement an iterator on top of pages. If all drivers support iterators natively, then it's probably fine to have that be the driver interface, but if not, I think it's more straightforward to use pages.

I don't see how this is any different than `List` for blob -- there also, GCS supports an iterator, but other drivers do not.